homeassistant/components/weather/__init__.py

```diff
@@ -110,6 +118,14 @@ def state_attributes(self):
         if attribution is not None:
             data[ATTR_WEATHER_ATTRIBUTION] = attribution
 
+        forecast = self.forecast
+        if forecast is not None:
+            for forecast_entry in forecast:
+                forecast_entry[ATTR_FORECAST_TEMP] = self._temp_for_display(
```

Comment:
Please make a copy of the data instead of mutating the returned data. `forecast_entry = dict(forecast_entry)`